Sources/StreamVideoSwiftUI/CallView/LayoutComponents/DominantSpeakerLayoutComponent.swift

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import StreamVideo
+import SwiftUI
+
+/// `DominantSpeakerLayoutComponent` represents a view for the dominant speaker's thumbnail.
+public struct DominantSpeakerLayoutComponent<Factory: ViewFactory>: View {
+
+    // MARK: - Properties
+
+    /// Factory for creating views.
+    public var viewFactory: Factory
+
+    /// The dominant speaker participant whose thumbnail will be displayed.
+    public var participant: CallParticipant
+
+    /// Suffix for constructing the view ID.
+    public var viewIdSuffix: String
+
+    /// Information about the call (if available).
+    public var call: Call?
+
+    /// Frame in which the dominant speaker's thumbnail will be displayed.
+    public var availableFrame: CGRect
+
+    /// Closure called to change visibility of the dominant speaker's track.
+    public var onChangeTrackVisibility: (CallParticipant, Bool) -> Void
+
+    /// Computed property to generate a unique view ID for the dominant speaker.
+    var viewId: String { participant.id + (!viewIdSuffix.isEmpty ? "-\(viewIdSuffix)" : "") }
```

Comment:
I assume the logic here was kept the same as before, right? Since these have a lot of impact on the visibility of the tracks.

Reply:
Exactly, this is the same. There is also a test case that confirms it. It's in this way so it can be reused between spotlight and screensharing